modules/openideal_user/src/EventSubscriber/OpenidealUserEventSubscriber.php

```diff
@@ -2,22 +2,36 @@
 
 namespace Drupal\openideal_user\EventSubscriber;
 
+use Drupal\Component\Datetime\Time;
+use Drupal\Core\Session\AccountProxy;
 use Drupal\ckeditor_mentions\CKEditorMentionEvent;
 use Drupal\comment\Entity\Comment;
 use Drupal\content_moderation\Event\ContentModerationEvents;
 use Drupal\content_moderation\Event\ContentModerationStateChangedEvent;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Url;
 use Drupal\openideal_user\Event\OpenidealContentModerationEvent;
 use Drupal\openideal_user\Event\OpenidealUserEvents;
 use Drupal\openideal_user\Event\OpenidealUserMentionEvent;
+use Drupal\user\UserDataInterface;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
 
 /**
  * Class OpenidealUserEventSubscriber.
  */
 class OpenidealUserEventSubscriber implements EventSubscriberInterface {
 
+  /**
+   * Tour related information.
+   */
+  const FRONT_PAGE_TOUR_COOKIE = 'OpenideaL_visitor_tour_front_page';
```

Comment:
1) Can we use the same name for both cases (cookie and data storage)?
2) Can we simplify it? Something like "tour_displayed".

Let's do not append "front_page" for now. In case of many tours, we'll need some generic solution (for example, append tour id to the parameter name).

Reply:
Done